src/combinatorics/simple_graph/basic.lean

```diff
@@ -363,6 +364,35 @@ begin
   exact ⟨v, by simp, finset.subset_univ _⟩,
 end
 
+lemma card_common_neighbors_le_degree (v w : V) :
```

Comment:
`card_common_neighbors_le_degree_left` maybe? And you can have a `le_degree_right` version which should be easy to prove from this